UnityProject/Assets/Scripts/Systems/Lobby/CharacterManager.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+using UnityEngine;
+
+namespace Systems.Character
+{
+	/// <summary>Manage a player's characters. Intended for the local player client.</summary>
+	public class CharacterManager
+	{
+		public List<CharacterSheet> Characters { get; } = new();
+
+		private string OfflineStoragePath => $"{Application.persistentDataPath}characters.json";
+
+		public CharacterManager Init()
+		{
+			LoadOfflineCharacters();
+
+			return this;
+		}
+
+		public int GetCurrentCharacterIndex()
+		{
+			int lastCharacterIndex = PlayerPrefs.GetInt(PlayerPrefKeys.LastCharacterIndex);
```

Comment:
perhaps this could be moved to the init and add a property to get the current selected index to the class. That way you can make it a ``BehaviourSubject``(https://github.com/unitystation/unitystation/blob/develop/UnityProject/Assets/Scripts/Util/Rx/BehaviourSubject.cs). You would then subscribe to it from the GUI and react to when it is changed.

Reply:
I don't like the current indexing and how this is set up will probably change later, but will keep this cool class in mind 👍